daffodil-runtime1/src/main/scala/org/apache/daffodil/runtime1/processors/DataProcessor.scala

```diff
@@ -501,7 +501,7 @@ class DataProcessor(
   }
 
   def unparse(inputter: InfosetInputter, output: DFDL.Output): DFDL.UnparseResult = {
-    Assert.usage(!this.isError)
+    Assert.usage(!this.isError, this.getDiagnostics.mkString)
```

Comment:
Grepping through the code, there are a ton of places where use have usage checks. I *think* all of the places except for the ones in this PR indicate a bug in Daffodil if they fail. However, the usage checks in this PR indicate a API user doing something wrong.

So we sort of have two different kinds of usage checks. One where we're doing internal Daffodil checks and one 
doing external API checks. I wonder if the former wants to stay as aborts, and only these user API ones want to become IllegalStateExceptions? Maybe using Assert.usage in the places in this PR is wrong and they should be something else? Or maybe all the internal ones want to be Assert.invariant? Or maybe we need two different kinds of usage functions, one for internal API and one for public API checks?